### DISCUSSION TITLE
Playing sound through AvAudioEngine in a .NET Maui calling app [iOS]

### DISCUSSION THREAD

**Discussion by Laursen1992** [dotnet/maui#17927] [Q&A]
My team and I are rewriting an app into .net Maui, in which it is possible to make phone calls using voip/softphone. Our old way of handling the in-call sound, resulted in some echo-problems when either one the participants used speakers on an Iphone. So we decided to look for alternatives to handling audio, and AvAudioEngine seemmed to offer a solution, which is also a part of .NET Maui. The only problem is the lack of documentation, as none of us have any real experience in this field.

We managed to figure out how to make the microphone/input part work, and are now trying to grasp how to output the sound through AvAudioEngine.

In short, we recieve the sound as bytes in short time intervals, copy it to IntPtr. Our best guess right now is to make a new AVAudioPcmBuffer, and fill it's AudioBufferList with that IntPtr, and use the PcmBuffer as parameter in prepareBuffer() method on a playerNode, and use the completionHandler to call the Play() on the playerNode, as we receive the audio, but so far without any luck. Is there anyone out there who knows how to handle this scenario? 

Thanks in advance!
